### PR TITLE
fix(security): validate allowed_token_ids against model output vocab size

### PR DESCRIPTION
**Pull request by jperezdealgaba**
Validate allowed_token_ids against model_config.get_vocab_size() instead of len(tokenizer) to prevent tokenizer-valid but output-vocab-invalid IDs from reaching the GPU logit-bias kernel.

For multimodal models (e.g., Ultravox), the tokenizer may contain input-only placeholder tokens whose IDs exceed the model's output vocabulary width. Using these IDs as logits column indices causes out-of-bounds access in the Triton _bias_kernel, enabling cross-request logits corruption and token allowlist bypass (GHSA-wr27-mx79-rg6c).

Changes:
- SamplingParams._validate_allowed_token_ids now uses model_config
- _bias_kernel adds per-ID bounds mask (defense-in-depth)
- Rust frontend validates against model_vocab_size
- Regression tests for validation and kernel bounds

### CODE REVIEW

**Comment by BugenZhao** [comment]
@codex review

File: rust/src/text/src/lower/token_ids.rs [hunk 1]
```diff
@@ -75,7 +75,7 @@ pub(crate) fn validate_vocab_range(
         validate_param(
             "allowed_token_ids",
             token_ids.iter().copied(),
-            limits.tokenizer_vocab_size,
+            limits.model_vocab_size,
```
[the remark this answers is not in the document]
Comment (jperezdealgaba) [reply rust/src/text/src/lower/token_ids.rs:78]: Just fixed this

**Review by BugenZhao — COMMENTED** [review]
LGTM on the Rust frontend side
